MachineLearning/ibmdbpy_NaiveBayes.ipynb

```diff
@@ -0,0 +1,957 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__Build the model__"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the model\n",
+    "NaiveBayesModel = learn.naive_bayes.NaiveBayes(modelname=\"NAIVE_IRIS\", disc = 'ef', bins = 10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_column_id', '_idadb', '_idadf', '_retrieve_NaiveBayes_Model', 'bins', 'coldefrole', 'coldeftype', 'colpropertiestable', 'describe', 'disc', 'fit', 'fit_predict', 'get_params', 'incolumn', 'labels_', 'mestimation', 'modelname', 'outtable', 'outtableProb', 'predict', 'set_params', 'target']\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "{'modelname': 'NAIVE_IRIS',\n",
+       " 'disc': 'ef',\n",
+       " 'bins': 10,\n",
+       " 'target': None,\n",
+       " 'incolumn': None,\n",
+       " 'coldeftype': None,\n",
+       " 'coldefrole': None,\n",
+       " 'colpropertiestable': None,\n",
+       " 'outtable': None,\n",
+       " 'outtableProb': None,\n",
+       " 'mestimation': None}"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Inspect the model\n",
+    "NaiveBayesModel.get_params()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All of the above parameters are optional. \n",
+    "\n",
+    "* modelname: If not specified, will be automatically generated. If a model with the same name already exists, it will be replaced.\n",
+    "\n",
+    "* disc: discretization type. Options are: Equal-frequency (ef), Minimal entropy (em), Equal-width (ew), Equal-width discretization with nice bucket limits (ewn).\n",
+    "\n",
+    "* bins: int, number of buckets for discretization of numeric columns\n",
+    "\n",
+    "* target: str, column of the input table that represents the class\n",
+    "\n",
+    "* incolumn: list, columns of the input table that have specific properties. If None, then all columns have default properties. Properties can be about the type (continuous or nominal) or the role(id, target, to ignore, input).\n",
+    "\n",
+    "More details on the [ibmdbpy documentation](https://pythonhosted.org/ibmdbpy/naive_bayes.html) and the [IBM Knowledge Center](https://www.ibm.com/support/knowledgecenter/SSPT3X_4.0.0/com.ibm.db2.luw.analytics.commsql.doc/doc/r_naive_bayes_procedure.html)"
```

Comment:
The Knowledge Center link leads to an old version of the BigInsights documentation. That's most likely both outdated and unmaintained.

```suggestion
    "More details on the [ibmdbpy documentation](https://pythonhosted.org/ibmdbpy/naive_bayes.html) and the [IBM Knowledge Center](https://www.ibm.com/support/knowledgecenter/SSCJDQ/com.ibm.swg.im.dashdb.analytics.doc/doc/r_naive_bayes.html)"
```